res/controllers/Arturia-KeyLab-Mk1-scripts.js

```diff
@@ -0,0 +1,1007 @@
+                midi: [0xB0, 0x49],
+                key: "volume",
+                softTakeover: softTakeover,
+            });
+            this.faders[1] = new components.Pot({
+                group: "[Channel1]",
+                midi: [0xB0, 0x4B],
+                key: "volume",
+                softTakeover: softTakeover,
+            });
+            this.faders[2] = new components.Pot({
+                group: "[Channel2]",
+                midi: [0xB0, 0x4F],
+                key: "volume",
+                softTakeover: softTakeover,
+            });
+            this.faders[3] = new components.Pot({
+                group: "[Channel4]",
+                midi: [0xB0, 0x48],
+                key: "volume",
+                softTakeover: softTakeover,
+            });
+            this.faders[4] = new components.Pot({
+                group: "[Channel3]",
+                midi: [0xB0, 0x50],
+                key: "rate",
+                softTakeover: softTakeover,
+            });
+            this.faders[5] = new components.Pot({
+                group: "[Channel1]",
+                midi: [0xB0, 0x51],
+                key: "rate",
+                softTakeover: softTakeover,
+            });
+            this.faders[6] = new components.Pot({
+                group: "[Channel2]",
+                midi: [0xB0, 0x52],
+                key: "rate",
+                softTakeover: softTakeover,
+            });
+            this.faders[7] = new components.Pot({
+                group: "[Channel4]",
+                midi: [0xB0, 0x53],
+                key: "rate",
+                softTakeover: softTakeover,
+            });
+            this.faders[8] = new components.Pot({
+                group: "[Master]",
+                midi: [0xB0, 0x55],
+                key: "crossfader",
+                softTakeover: softTakeover,
+            });
+            // Pots
+            this.mainGain = new components.Pot({
+                group: "[Master]",
+                midi: [0xB0, 0x07],
+                key: "gain",
+                softTakeover: softTakeover,
+            });
+            this.headGain = new components.Pot({
+                group: "[Master]",
+                midi: [0xB0, 0x00],
+                key: "headGain",
+                softTakeover: softTakeover,
+            });
+        }
+    }
+
+    KeyLabMk1.init = function(id, _debugging) {
+        KeyLabMk1.controller = new Controller(id);
+        writeDisplay("Welcome to Mixxx");
+    };
+
+    KeyLabMk1.shutdown = function() {
+        KeyLabMk1.controller.shutdown();
+        writeDisplay("Goodbye!");
+    };
+
+    KeyLabMk1.incomingData = function(data, length) {
+        if (length < 6) {
+            console.log(`expected sysex packet of length 6, got ${length}`);
+            return;
+        }
+        for (let n = 0; n < MMCHeader.length; n++) {
+            if (data[n] !== MMCHeader[n]) {
+                console.log("unknown sysex packet");
```

Comment:
NIT: consider logging the packet too. that way the logs would be more useful by default.